@kiva/kv-components/vue/KvLightbox.vue

```diff
@@ -0,0 +1,288 @@
+						:role="role"
+						@click.stop
+					>
+						<!-- header -->
+						<div
+							class="
+								tw-flex
+								tw-p-2.5 md:tw-px-4 md:tw-pt-4 md:tw-pb-3.5
+							"
+						>
+							<div>
+								<!-- @slot header -->
+								<slot name="header">
+									<h2	class="tw-text-h3 tw-flex-1">
+										{{ title }}
+									</h2>
+								</slot>
+							</div>
+							<button
+								v-if="!preventClose"
+								class="
+									tw-grid tw-content-center tw-justify-center
+									tw-ml-auto
+									tw-w-6 tw-h-6 tw--m-2
+									hover:tw-text-action-700
+								"
+								@click.stop.prevent="hide"
+							>
+								<kv-material-icon
+									class="tw-w-3 tw-h-3"
+									:icon="mdiClose"
+								/>
+								<span class="tw-sr-only">Close</span>
+							</button>
+						</div>
+
+						<!-- body -->
+						<div
+							id="kvLightboxBody"
+							ref="kvLightboxBody"
+							class="
+								tw-flex-1
+								tw-px-2.5 md:tw-px-4
+								tw-pb-2.5 md:tw-pb-4
+								tw-overflow-auto
+							"
+						>
+							<!-- @slot default -->
+							<slot>Lightbox body</slot>
+						</div>
+
+						<!-- controls -->
+						<div
+							v-if="$slots.controls"
+							ref="controlsRef"
+							class="
+								tw-flex-shrink-0 tw-flex tw-justify-end tw-gap-x-2.5
+								tw-p-2.5 md:tw-px-4 md:tw-pb-4 md:tw-pt-1
+							"
+						>
+							<!-- @slot controls -->
+							<slot name="controls"></slot>
+						</div>
+					</div>
+				</div>
+			</focus-lock>
+		</div>
+	</transition>
+</template>
+
+<script>
+
+import { mdiClose } from '@mdi/js';
+import FocusLock from 'vue-focus-lock';
+import { hideOthers as makePageInert } from 'aria-hidden';
+import { lockScroll, unlockScroll } from '../utils/scrollLock';
+import { lockPrintSingleEl, unlockPrintSingleEl } from '../utils/printing';
+
+import KvMaterialIcon from './KvMaterialIcon.vue';
+
+/**
+ * Alert or a lightbox
+ * Accesibility: https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal
+ *
+ * - [x] Tab and Shift + Tab do not move focus outside the dialog
+ * - [x] focus is initially set on the first focusable element (close button).
+ * - [x] focus is returned to the element that opened the dialog on close
+ * - [x] role = dialog
+ * - [x] aria-label is set to its title.
+ * - [x] adds aria-hidden=true to all elements other than this dialog when open. - https://github.com/theKashey/vue-focus-lock/issues/16
+ * - [x] scrolling only scrolls the lightbox contents, not the page itself
+
+ * Alert dialog - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alertdialog_role
+ *
+ * - [x] focus moves to the first non-destructive control, rather than the close button
+ * - [x] role = alertDialog
+ * - [x] aria-describedby is set to the id of the dialog body
+ *
+ * Printing
+ *
+ * - [x] Only prints the contents of the lightbox when open
+ * TODO: ensure purge doesn't remove these classes
```

Comment:
@emuvente, @mcstover   Any ideas on how to keep these classes from being purged? They get added by a .js util.  I tried adding them to a `safelist` in the shared tailwind config purge object, but that will get overridden by the project tailwind config. As in, you can't "extend" the purge part of the config.

One way would be a blank div like `<div class="tw-hidden print:tw-invisible print:tw-visible..."></div>` in this file's template. Pragmatic but not super clean.

Reply:
When including kv-components in other projects, we already need to ensure that the vue files from kv-components are included in the purge options, so we could include directions in the README about which purge options to be sure to use when adding kv-components to a project.

Your suggestion of an invisible div with the classes is not the worst alternative though.

Reply:
Reviewing the docs it seems like the syntax changes when introducing more specific purge configuration. For example when setting up "safeList" options they live alongside a "content" array. We currently don't purge anything (and have not specified which files to include/exclude) in kv-components so adding the safeList without corresponding configs just purged everything except the print styles.

Thus, I think we could add the safeList for print styles in the UI repo as Shua mentioned. We might also need to change our syntax there to have the "contents" array alongside the "safeList" array instead of just a root array in our tailwind config.

Reply:
I spun up a simple Vue cli app + tailwind  and did some various tests with purge. It's very lenient. Classes written in the `<script>` tag of .vue files get preserved, like 
```
addAClass() { this.classList.add('tw-float-left') }
```
Commented out classes get picked up, like
```
<!-- <div class="tw-float-right"></div> -->
```
 Basically if it finds any possible tailwind class anywhere in any file it seems to preserve it from being purged.

Reply:
For this PR we're adding tailwind classes from `utils/printing.js` and `utils/scrollLock.js`, which won't get pulled in by the current purge settings on UI. So I think we just need to add `'./node_modules/@kiva/kv-components/utils/**/*.js'` to the list on UI and we're set.

I'll make a separate PR explaining this in the README.